src/utils/starknet_compilation.py

```diff
@@ -72,3 +73,18 @@ def compile_contract(
         )
         assert isinstance(assembled, ContractDefinition)
         return assembled
+
+    class AbiElement(TypedDict):
+        name: str
+        type: Literal["function"]
+        inputs: List[Any]
+        outputs: List[Any]
+        stateMutability: NotRequired[str]
+
+    def get_functions(self, cairo_file_path: Path, prefix: str) -> List[AbiElement]:
+        preprocessed = self.preprocess_contract(cairo_file_path)
```

Comment:
It would better fit inside TestCase class

Reply:
Probably we want to move compilation to collector in the future (to prevent changing cairo code when tests are running), so keep that in mind, this desing will make this change a bit hard.

Reply:
I think you meant `TestSubject`. It's actually a better class for this method.

`Probably we want to move compilation to collector in the future (to prevent changing cairo code when tests are running), so keep that in mind, this desing will make this change a bit hard.`

Collector shouldn't work directly on `StarknetPreprocessedProgram`. Compiling at the very beginning will increase initial delay, and we're already dealing with this problem. I am not sure whether other test runners allow modifying test files during running tests. Maybe we should simply copy files when the test command starts.

Reply:
I tried to move `get_functions` to `TestSubject` and I changed my mind. `get_functions` should remain in the `StarknetCompiler` to avoid exposing `StarknetPreprocessedProgram` to other classes. 

`StarknetCompiler` is a facade for `StarknetPreprocessedProgram`. It's useful to use facades, especially if the external library often changes.  We may create a new class —  `StarknetPreprocessedProgramFacade` in the future, if necessary.